lib/vector-buffers/src/variants/disk_v2/common.rs

```diff
@@ -291,3 +308,34 @@ where
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use proptest::{prop_assert, prop_assert_eq, proptest};
+
+    use crate::variants::disk_v2::{common::DEFAULT_MAX_DATA_FILE_SIZE, DiskBufferConfigBuilder};
+
+    const MAX_BUFFER_SIZE_TEST_UPPER_BOUND_INPUT: usize = DEFAULT_MAX_DATA_FILE_SIZE * 10;
+
+    proptest! {
+        #[test]
+        fn ensure_max_disk_buffer_size_lower_bound(max_buffer_size in DEFAULT_MAX_DATA_FILE_SIZE..MAX_BUFFER_SIZE_TEST_UPPER_BOUND_INPUT) {
+            // This is a little ugly but the `u64`/`usize` conversions make it annoying to do in a full const way.
+            let default_max_data_file_size = u64::try_from(DEFAULT_MAX_DATA_FILE_SIZE)
+                .expect("`DEFAULT_MAX_DATA_FILE_SIZE` should not ever be greater than `u64::MAX`.");
+            let max_buffer_size = u64::try_from(max_buffer_size)
+                .expect("`max_buffer_size` should not ever be greater than `u64::MAX`.");
+
+            let config = DiskBufferConfigBuilder::from_path("/tmp/dummy/path")
```

Comment:
Should this be generated with `temp_dir()`?

Reply:
Nothing is actually being written or checked for existence, so we just need a value that is valid from a "valid UTF-8 characters" sort of perspective in order to satisfy the bounds.